perfkitbenchmarker/disk.py

```diff
@@ -26,10 +26,29 @@
 PIOPS = 'piops'  # Provisioned IOPS (SSD) in AWS
 LOCAL = 'local'
 
+_DISK_SPEC_REGISTRY = {}
+
+
+def GetDiskSpecClass(cloud):
+  """Get the DiskSpec class corresponding to 'cloud'."""
+  return _DISK_SPEC_REGISTRY.get(cloud, BaseDiskSpec)
+
+
+class AutoRegisterDiskSpecMeta(type):
+  """Metaclass which automatically registers DiskSpecs."""
+
+  def __init__(cls, name, bases, dct):
+    if cls.CLOUD:
```

Comment:
I now raise an exception if cls.CLOUD has already been registered (which is equivalent to what you asked for because BaseDiskSpec will be registered first with CLOUD == None).